prog/dftb+/lib_dftbplus/main.F90

```diff
@@ -3623,6 +3629,27 @@ subroutine getNextInputCharges(env, pChrgMixer, qOutput, qOutRed, orb, nIneqOrb,
       #:endif
         call expandCharges(qInpRed, orb, nIneqOrb, iEqOrbitals, qInput, dftbU, qBlockIn,&
             & iEqBlockDftbu, species0, qiBlockIn, iEqBlockDftbuLS, iEqBlockOnSite, iEqBlockOnSiteLS)
+
+        if (isAContactCalc) then
+          ! symmetrize charges between the principle layers of the contact
+          do iAt = 1, nAtom/2
+            qInput(:,iAt,:) = 0.5_dp*(qInput(:,iAt,:) + qInput(:,iAt+nAtom/2,:))
+            qInput(:,iAt+nAtom/2,:) = qInput(:,iAt,:)
+          end do
+          if (allocated(qBlockIn)) then
+            do iAt = 1, nAtom/2
+              qBlockIn(:,:,iAt,:) = 0.5_dp*(qBlockIn(:,:,iAt,:) + qBlockIn(:,:,iAt+nAtom/2,:))
+              qBlockIn(:,:,iAt+nAtom/2,:) = qBlockIn(:,:,iAt,:)
+            end do
+          end if
+          if (allocated(qiBlockIn)) then
+            do iAt = 1, nAtom/2
+              qiBlockIn(:,:,iAt,:) = 0.5_dp*(qiBlockIn(:,:,iAt,:) + qiBlockIn(:,:,iAt+nAtom/2,:))
+              qiBlockIn(:,:,iAt+nAtom/2,:) = qiBlockIn(:,:,iAt,:)
+            end do
+          end if
+        end if
```

Comment:
You might consider to put this into a separate subroutine (e.g. `symmetrizeContactCharges()`) and call it from here.